pandera/backends/polars/builtin_checks.py

```diff
@@ -249,26 +230,26 @@ def str_endswith(data: PolarsData, string: str) -> pl.LazyFrame:
 )
 def str_length(
     data: PolarsData,
-    min_value: int = None,
-    max_value: int = None,
+    min_value: Optional[int] = None,
+    max_value: Optional[int] = None,
 ) -> pl.LazyFrame:
     """Ensure that the length of strings is within a specified range.
 
     :param data: NamedTuple PolarsData contains the dataframe and column name for the check. The keys
                 to access the dataframe is "dataframe" and column name using "key".
-    :param min_value: Minimum length of strings (default: no minimum)
-    :param max_value: Maximum length of strings (default: no maximum)
+    :param min_value: Minimum length of strings (including) (default: no minimum)
+    :param max_value: Maximum length of strings (including) (default: no maximum)
     """
     # NOTE: consider using len_bytes (faster but returns != n_chars for non ASCII strings
-    n_chars = pl.col("string_col").str.n_chars()
+    n_chars = pl.col(data.key).str.n_chars()
     is_in_min = (
         n_chars.ge(min_value) if min_value is not None else pl.lit(True)
     )
     is_in_max = (
         n_chars.le(max_value) if max_value is not None else pl.lit(True)
     )
 
-    return data.dataframe.select(is_in_min.and_(is_in_max))
+    return data.dataframe.select(is_in_min.and_(is_in_max).alias(data.key))
```

Comment:
We need to explicitly alias here. If `min_value` is None, then resulting column name will be 'literal', which will break Polars lazy computation. Seems to be related to https://github.com/pola-rs/polars/issues/14382

Reply:
Would this make sense to do instead:

```python
if min_value is None and max_value is None:
    raise ValueError("Must provide at least on of 'min_value' and 'max_value'")

n_chars = pl.col(data.key).str.n_chars()
if min_value is None:
    expr = n_chars.le(max_value)
elif max_value is None:
    expr = n_chars.ge(min_value)
else:
    expr = n_chars.between(min_value, max_value)

return expr
```